From d1da8cd20554aa1d3c766855c251a5a643d2abb4 Mon Sep 17 00:00:00 2001
From: Jonas Wloka <code@jowlo.de>
Date: Sat, 17 Nov 2018 19:54:20 +0100
Subject: [PATCH] Filter out boxes that start at (0, 0)

Tesseract returns way too large boxes that cover the whole page, mostly
containing only a single special character. All of these boxes (in my
tests) have coordinate 0 0.

This filters out all of these boxes.
---
 paperwork-gtk/src/paperwork/frontend/mainwindow/pages.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/paperwork-gtk/src/paperwork/frontend/mainwindow/pages.py b/paperwork-gtk/src/paperwork/frontend/mainwindow/pages.py
index 35c6b7f4..e324c670 100644
--- a/paperwork-gtk/src/paperwork/frontend/mainwindow/pages.py
+++ b/paperwork-gtk/src/paperwork/frontend/mainwindow/pages.py
@@ -251,10 +251,14 @@ class JobPageBoxesLoader(Job):
             boxes = set()
             for line in line_boxes:
                 for word in line.word_boxes:
-                    if word.content.strip() == "":
+                    if word.content.strip() == "" or (word.position[0][0] == 0
+                                                      and word.position[0][1] == 0):
                         # XXX(Jflesch): Tesseract 3.03 (hOCR) returns big and
                         # empty word boxes sometimes (just a single space
                         # inside). They often match images, but not always.
+                        # XXX(jowlo): Tesseract returns large boxes containing
+                        # single letters (mostly special chars) that cover the
+                        # whole page. All of these start at (0 0)
                         continue
                     boxes.add(word)
 
-- 
2.20.0

